src/Chapter2.hs

```diff
@@ -728,7 +738,9 @@ value of the element itself
 🕯 HINT: Use combination of 'map' and 'replicate'
 -}
 smartReplicate :: [Int] -> [Int]
-smartReplicate l = error "smartReplicate: Not implemented!"
+smartReplicate [] = []
```

Comment:
This is redundant if you go with `concatMap` solution, as it handles all lists 🙂